near-sdk/src/store/frangible_map/mod.rs

```diff
@@ -18,23 +18,28 @@ use super::{FreeList, LookupMap, ERR_INCONSISTENT_STATE, ERR_NOT_EXIST};
 
 /// A lazily loaded storage map that stores its content directly on the storage trie.
 /// This structure is similar to [`near_sdk::store::LookupMap`](crate::store::LookupMap), except
-/// that it stores the keys so that [`UnorderedMap`] can be iterable.
+/// that it stores the keys so that [`FrangibleUnorderedMap`] can be iterable.
 ///
 /// This map stores the values under a hash of the map's `prefix` and [`BorshSerialize`] of the key
 /// using the map's [`ToKey`] implementation.
 ///
-/// The default hash function for [`UnorderedMap`] is [`Sha256`] which uses a syscall
+/// The default hash function for [`FrangibleUnorderedMap`] is [`Sha256`] which uses a syscall
 /// (or host function) built into the NEAR runtime to hash the key. To use a custom function,
 /// use [`with_hasher`]. Alternative builtin hash functions can be found at
 /// [`near_sdk::store::key`](crate::store::key).
 ///
+/// # Performance considerations
+/// Note that this collection is optimized for fast removes at the expense of key management.
+/// If the amount of removes is significantly higher than the amount of inserts the iteration
+/// becomes more costly. See [`remove`](FrangibleUnorderedMap::remove) for details.
```

Comment:
As a developer I am left with an unanswered question in my head: "and now what? are there better options?"

Reply:
Fixed.